src/engine/controls/loopingcontrol.cpp

```diff
@@ -136,6 +136,29 @@ LoopingControl::LoopingControl(const QString& group,
             this,
             [this](double value) { slotBeatLoop(value); },
             Qt::DirectConnection);
+    m_pCOBeatLoopAnchor = new ControlObject(ConfigKey(group, "beatloop_anchor"),
+            true,
+            false,
+            false,
+            static_cast<double>(LoopAnchorPoint::Start));
+    m_pCOBeatLoopAnchor->connectValueChangeRequest(this,
+            &LoopingControl::slotBeatLoopAnchorChangeRequest,
+            Qt::DirectConnection);
+
+    m_pCOBeatLoopAnchorToggle = new ControlObject(
+            ConfigKey(group, "beatloop_anchor_toggle"), false);
+    connect(
+            m_pCOBeatLoopAnchorToggle,
+            &ControlObject::valueChanged,
+            this,
+            [this](double value) {
+                if (value > 0) {
+                    slotBeatLoopAnchorChangeRequest(static_cast<double>(
```

Comment:
I think you can read the value of m_pCOBeatLoopAnchorToggle right away using `toBool()` without any connection.

Reply:
I've used `toBool()` but not sure how I can make the `toggle` work without connection. Could yous how me a example?

Reply:
You need to make the control a toggle pushbutton like here: https://github.com/mixxxdj/mixxx/blob/7b3e48544a5a5475553742678115951179c41526/src/skin/skinloader.cpp#L233

Reply:
Sorry, I'm not sure how you want me to remove the connection. This should already be a `ControlPushButton`. In the example you've linked, there also seem to be a `connect`

```cpp
m_pSelectBigSpinnyCover = std::make_unique<ControlPushButton>(
            ConfigKey("[Skin]", "select_big_spinny_or_cover"), true);
m_pSelectBigSpinnyCover->setButtonMode(ControlPushButton::TOGGLE);
connect(m_pSelectBigSpinnyCover.get(),
            &ControlObject::valueChanged,
            this,
            &SkinLoader::updateSpinnyCoverControls);
```

Just to make sure you haven't missed it, but this `connect` is for the `toggle` (`loop_anchor_toggle`)  control, which flip the state of the main CO (`loop_anchor`), later use in the loop logic. Do you think I can still remove the connect for the toggle? If yes, could you make a code suggestion?

Reply:
The toggling is handled inside the control, via ControlPushButton::TOGGLE 
You need only to connect the signal when you like to react on the event. But this is not the case here.

Reply:
Why do I not want to react to the toggle event here?  Since the loop anchor state is stored in an other CO, do I not need to replicate it here? I'm confused

Reply:
I think you can do it with just with one CO. But maybe I miss something?